### ISSUE TITLE
Incorrect input transformation when changing transfomers

### ISSUE THREAD

**Issue by nimobeeren** [SaraVieira/css-to-js#21]
## Steps to reproduce
1. Start with the default input:
```
display: block;
font-size: 16px;
background: #1e2f5d;
color: #a4cff4;
font-family: "Inter", sans-serif;
font-weight: bold;
```
2. Change transformer to "React props => JS object"
3. Change transformer to "CSS => JS object"
4. Observe new input:
```
16: true;
display: true;
block: true;
font: true;
size: true;
background: true;
1e2f5d: true;
color: true;
a4cff4: true;
family: true;
-inter: true;
sans: true;
serif: true;
weight: true;
bold: true;
```

## Expected behaviour
The expected input would be the same as the original input:
```
display: block;
font-size: 16px;
background: #1e2f5d;
color: #a4cff4;
font-family: "Inter", sans-serif;
font-weight: bold;
```

## Additional details
Bug was introduced in: #19 

The transformation "React props => CSS" seems to work fine with that input. That would suggest that the problem is not in that transformer.

**Comment by SaraVieira** [comment]
Niiice!

**Comment by nimobeeren** [comment]
I've implemented a fix in #24.

However, I still don't quite understand the bug completely. Through debugging I went through the following steps:

1. default transformer is "CSS => JS"
2. change transformer to "JSX => JS"
3. change transformer back to "CSS => JS"

On step 3, I found that the reducer for setting state of `transformer` ran twice. And this only happened when switching from "JSX => JS" to something else, for all other transformers it just ran once. Here's [the affected reducer](https://github.com/SaraVieira/css-to-js/blob/77a4f8c0d1fadea9fe7db29a6cc70175b1517a89/src/pages/home.tsx#L26).

With the fix in place, that reducer has essentially been replaced by a `useEffect` that handles transforming input when the transformer changes. The effect does seem to run the same way for every transformer. Here's [the effect from the fix](https://github.com/SaraVieira/css-to-js/blob/fbe1387522c47f1520512096cc6b74db53a09c98/src/pages/home.tsx#L24).
